lib/livebook_web/live/session_helpers.ex

```diff
@@ -21,19 +24,36 @@ defmodule LivebookWeb.SessionHelpers do
 
     case Livebook.Sessions.create_session(opts) do
       {:ok, session} ->
-        push_redirect(socket, to: Routes.session_path(socket, :page, session.id))
+        redirect_path = session_path(socket, session, opts)
+        push_redirect(socket, to: redirect_path)
 
       {:error, reason} ->
         put_flash(socket, :error, "Failed to create session: #{reason}")
     end
   end
 
+  @doc """
+  Generate the session path based on the provided options.
+  """
+  @spec session_path(Socket.t(), Session.t(), keyword()) :: String.t()
+  def session_path(socket, %Session{id: session_id}, opts \\ []) do
```

Comment:
Since we only need an id, let's pass only that as an argument, in case we want to use it elsewhere, where we only have the id :)